### ISSUE TITLE
Ability to hook into and customize data download and distribution

### ISSUE THREAD

**Issue by cdahlqvist** [elastic/rally#335]
Rally currently handles download, decompression, and partitioning of data files automatically in the background. It would be very useful to have hooks into this process so that it can be overridden or customized for specific tracks. I do not necessarily see any need to customize the download and decompression, but it would be useful if it was possible to perform some level of preprocessing on the files before they are being used and also control how the data is distributed across worker threads.

**Comment by cdahlqvist** [comment]
I realized it may be useful to add a bit more context around this feature.

The built-in functionality to download and extract data files is very useful but does not offer a lot of flexibility around how to use and distribute the data. If we were able to hook into this process at various stages it would be possible to e.g. pre-process the downloaded and extracted file and break up records into parts that allows real-time timestamps to efficiently be inserted while using this data as a template. This would allow us to avoid wasting CPU cycles on parsing, altering and serializing each event repeatedly.
